kohttp-test/src/test/kotlin/io/github/rybalkinsd/kohttp/dsl/HttpDslKtTest.kt

```diff
@@ -0,0 +1,209 @@
+                cookie {
+                    "aaa" to "bbb"
+                    "ccc" to 42
+                }
+            }
+
+            body {
+                form {
+                    "login" to "user"
+                    "email" to "john.doe@gmail.com"
+                }
+            }
+        }).use {
+            val parsedResponse = it.toJson()
+            assertContainsAtLeast(expectedHeader, parsedResponse["headers"])
+            assertContainsExactly(expectedParams, parsedResponse["args"])
+            assertContainsExactly(expectedForm, parsedResponse["form"])
+            assertThat(it.code()).isEqualTo(200)
+        }
+    }
+
+    @Test
+    fun `get request with param and header`() {
+        val variable = 123L
+        val expectedHeader = mapOf(
+                "one" to "42",
+                "two" to variable.toString(),
+                "three" to """{"a":$variable,"b":{"b1":"512"},"c":[1,2.0,3]}""",
+                "cookie" to "aaa=bbb; ccc=42"
+        )
+
+        val expectedParams = mapOf(
+                "text" to "iphone",
+                "lr" to "213"
+        )
+
+        val response = http(method = Method.GET) {
+            host = "postman-echo.com"
+            path = "/get"
+
+            header {
+                "one" to 42
+                "two" to variable
+                "three" to json {
+                    "a" to variable
+                    "b" to json {
+                        "b1" to "512"
+                    }
+                    "c" to listOf(1, 2.0, 3)
+                }
+
+                cookie {
+                    "aaa" to "bbb"
+                    "ccc" to 42
+                }
+            }
+
+            param {
+                "text" to "iphone"
+                "lr" to 213
+            }
+        }
+        response.use {
+            val parsedResponse = it.toJson()
+            assertContainsAtLeast(expectedHeader, parsedResponse["headers"])
+            assertContainsExactly(expectedParams, parsedResponse["args"])
+            assertThat(it.code()).isEqualTo(200)
+        }
+    }
+
+    @Test
+    fun `get request with body throws exception`() {
+        val variable = 123L
+
+        assertThatCode {
+            http(method = Method.GET) {
+                host = "postman-echo.com"
+                path = "/get"
+
+                header {
+                    "one" to 42
+                    "two" to variable
+                    "three" to json {
+                        "a" to variable
+                        "b" to json {
+                            "b1" to "512"
+                        }
+                        "c" to listOf(1, 2.0, 3)
+                    }
+
+                    cookie {
+                        "aaa" to "bbb"
+                        "ccc" to 42
+                    }
+                }
+
+                param {
+                    "text" to "iphone"
+                    "lr" to 213
+                }
+
+                body {
```

Comment:
As I mentioned in c comment section, we should limit our support (at least in our first release for this feature)
So let's use only `HttpContext.() -> Unit`

Reply:
Yes, sounds good. I changed it back to only support HttpContext